### DISCUSSION TITLE
Kitty terminal graphics protocol: cursor disappeared when exit awrit

### DISCUSSION THREAD

**Discussion by Parsifa1** [ghostty-org/ghostty#2732] [Q&A] [answered]
```
Ghostty 0.1.0-main+29c3a52e
Darwin Kernel Version 24.1.0
```
when i exit [awrit](https://github.com/chase/awrit) with ctrl-c, cursor wil dispeared, and this can be fixed with `tput reset`.
i tried with kitty, but everything works wellk, so i guess maybe something wrong in kitty graphics protocol in ghostty?

before:
![Image](https://github.com/user-attachments/assets/50e8da8b-86c8-4d5e-8442-6732ffcfe1fc)
after:
![Image](https://github.com/user-attachments/assets/17b9fe1f-57b1-4990-9fac-04e7cab9ba0d)

kitty:
![Image](https://github.com/user-attachments/assets/5af34967-20e0-4fc4-a494-646f37e12f93)

relative:
https://github.com/chase/awrit/issues/20

**Comment by Parsifa1** [comment]
sorry for have sent in the wrong place :(
anyway it's an amazing terminal simulator, works perfect with least config (really out-of-box)

this is not a really big deal, just want to mark if is there something wrong in it

**Comment by rockorager** [comment, marked as answer]
I found what is causing this.

awrit uses a specific Kitty extension to save / restore *all* private modes. They rely on this to show the cursor after exiting:

>They save private modes (kitty does this as an extension which saves all private modes, but the standard is to only save specific private modes)
>
>They hide the cursor
>
>🪄 magic 🪄 
>
>They restore all private modes (which would restore the cursor, if you have that specific kitty extension)

**Comment by rockorager** [comment]
Fixed in: https://github.com/chase/awrit/pull/50
